cognite/client/_api/datapoints.py

```diff
@@ -600,16 +603,16 @@ def __init__(self, config: ClientConfig, api_version: Optional[str], cognite_cli
     def retrieve(
         self,
         *,
-        id: Union[None, int, Dict[str, Any], Sequence[Union[int, Dict[str, Any]]]] = None,
-        external_id: Union[None, str, Dict[str, Any], Sequence[Union[str, Dict[str, Any]]]] = None,
-        start: Union[int, str, datetime, None] = None,
-        end: Union[int, str, datetime, None] = None,
-        aggregates: Union[str, List[str], None] = None,
+        id: None | int | dict[str, Any] | Sequence[int | dict[str, Any]] = None,
+        external_id: None | str | dict[str, Any] | Sequence[str | dict[str, Any]] = None,
+        start: int | str | datetime | None = None,
+        end: int | str | datetime | None = None,
+        aggregates: AGGREGATE | List[AGGREGATE] | None = None,
```

Comment:
I agree that this is better, but changing this from `str` to `Literal` makes the type more strict and may break clients using a static type checker. If they've wrapped this method, and are passing a variable of type `str` it will start failing. e.g.
```
def my_dps_wrapper(id: int, ..., aggregate: str) -> Datapoints:
    return cognite_client.datapoints.retrieve(id=id, ..., aggregate=aggregate)

res = my_dps_wrapper(id=123, ..., aggregate="max")
```
is perfectly fine at runtime, but will not pass static type checking

Reply:
It is a good point, what about doing a union? Then, it is clear what aggregates exists, and at the same time we do not break static type checkers.
```suggestion
        aggregates: Union[AGGREGATE, str] | List[AGGREGATE | str] | None = None,
```